### ISSUE TITLE
"noalloc" deprecation warnings when building

### ISSUE THREAD

**Issue by cfcs** [mirage/ocaml-cstruct#154]
FWIW

```
File "lib/cstruct.ml", line 160, characters 0-138:
Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
File "lib/cstruct.ml", line 162, characters 0-132:
Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
File "lib/cstruct.ml", line 164, characters 0-132:
Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
File "lib/cstruct.ml", line 166, characters 0-132:
Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
File "lib/cstruct.ml", line 168, characters 0-132:
Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
File "lib/cstruct.ml", line 170, characters 0-117:
Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
File "lib/cstruct.ml", line 172, characters 0-102:
Warning 3: deprecated: [@@noalloc] should be used instead of "noalloc"
```

**Comment by avsm** [comment]
Thanks -- we need to bump up the minimum supported version of OCaml to 4.03.0, which is reasonable nowadays given that the rest of MirageOS' support requires that as well.

**Comment by avsm** [comment]
fixed in #157 and will be in 3.1.0
